### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by nickkolok**
If you want a smaller portion of commits, then just write me your requirements.

### CODE REVIEW

File: ipt_connect/IPTdev/models.py [hunk 1]
```diff
@@ -27,31 +27,21 @@ def mean(vec):
...
+	nreject = round(len(vec) / 4.0)
 
-	if round(nreject / 2.0) == nreject / 2.0:
-		nlow = int(nreject / 2.0)
-		nhigh = int(nlow)
-	else:
+	if nreject % 2:
 		nlow = int(nreject / 2.0 + 0.5)
-		nhigh = int(nreject / 2.0 - 0.5)
+		nhigh = -int(nreject / 2.0 - 0.5) or None
```
Comment (vbonvin) [inline ipt_connect/IPTdev/models.py]: I don't really understand that part...  -int(nreject / 2.0 - 0.5) will always be a defined value, why the "or None" statement?

File: ipt_connect/IPTdev/models.py [hunk 2]
```diff
@@ -27,31 +27,21 @@ def mean(vec):
 		return 0
 
 def special_mean(vec):
-	if len(vec) in [5, 6]:
-		nreject = 1
-	elif len(vec) in [7, 8]:
-		nreject = 2
-	else:
-		nreject = round(len(vec) / 4.0)
+	nreject = round(len(vec) / 4.0)
```
Comment (vbonvin) [inline ipt_connect/IPTdev/models.py]: For len(vec) = 6, this returns nreject=2 which is not as the previous behaviour

File: ipt_connect/IPTdev/models.py [hunk 3]
```diff
@@ -38,10 +38,16 @@ def special_mean(vec):
...
 	vec = vec[nlow:nhigh]
 
-	try:
+	if len(vec):
 		return float(sum(vec)) / len(vec)
-	except ZeroDivisionError:
-		return 0
+	return 0
+
+def iypt_mean(vec):
```
Comment (vbonvin) [inline ipt_connect/IPTdev/models.py]: Why implementing iypt_mean here? Is that used in national tournaments?

File: ipt_connect/IPTdev/parameters.py [hunk 4]
```diff
@@ -2,7 +2,10 @@
 
 # Views parameters
 app_version = "dev"     # keyword for url parsing
-NAME = "IPT dev"
+NAME = {
```
Comment (vbonvin) [inline ipt_connect/IPTdev/parameters.py]: hum...names should be generic in IPTdev :)

File: ipt_connect/IPTdev/parameters.py [hunk 5]
```diff
@@ -1,7 +1,16 @@
 # -*- coding: utf-8 -*-
 
 # Views parameters
-app_version = "dev"     # keyword for url parsing
+app_version = "IPTdev"[3:]     # keyword for url parsing
```
Comment (vbonvin) [inline ipt_connect/IPTdev/parameters.py]: That's indeed a dirty hack.

File: ipt_connect/IPTdev/views.py [hunk 6]
```diff
@@ -488,4 +488,4 @@ def rank_ordinal(value):
 				team.emphase=True
 			rankteamsB.append(team)
 
-	return render(request, 'IPT%s/poolranking.html' % params.app_version, {'rankteamsA': rankteamsA, 'rankteamsB': rankteamsB, 'name': params.NAME})
\ No newline at end of file
+	return render(request, 'IPT%s/poolranking.html' % params.app_version, {'rankteamsA': rankteamsA, 'rankteamsB': rankteamsB, 'params': params})
```
Comment (vbonvin) [inline ipt_connect/IPTdev/views.py]: still don't see the point of doing this, but I think it's harmless so OK

File: ipt_connect/IPTdev/models.py (hunk 2 above)
Comment (nickkolok) [reply ipt_connect/IPTdev/models.py]: Thank you, you ar right; I should revert this to the explicit one. According to the rules:

> If there are five or six jury members the lowest mark is discarded, if there are seven or eight jury members one lowest and one highest mark is discarded. For larger jury numbers, the number of discarded marks is quarter that of the jury members, rounded up (if this number is odd, one more of the lowest marks is discarded than of the highest).

It gives the following:
```
5 6 7 8 9 10 11 12
1 1 2 2 3  3  3  3
```

Am I right?

File: ipt_connect/IPTdev/models.py (hunk 3 above)
Comment (nickkolok) [reply ipt_connect/IPTdev/models.py]: Yes, we tried to use `ipt_connect` on sub-sublocal stage of IYPT (Repnoe). I cannot claim that the experiment was 100% successful, but we've made some onclusions and applied some patches.

File: ipt_connect/IPTdev/models.py (hunk 2 above)
Comment (vbonvin) [reply ipt_connect/IPTdev/models.py]: Yes. Basically the previous implementation did the job correcly, so you can simply revert to it

File: ipt_connect/IPTdev/parameters.py (hunk 5 above)
Comment (nickkolok) [reply ipt_connect/IPTdev/parameters.py]: It's waiting for a hero to be fixed. Looks like a good issue for a newbie (much work, few difficulties). See #77.

File: ipt_connect/IPTdev/parameters.py (hunk 4 above)
Comment (nickkolok) [reply ipt_connect/IPTdev/parameters.py]: Fixed: https://github.com/nickkolok/ipt_connect/commit/211b3682af1dfb20eb78daaa02a34ffaf8ec27e6

File: ipt_connect/IPTdev/models.py (hunk 1 above)
Comment (nickkolok) [reply ipt_connect/IPTdev/models.py]: Reverting the whole function to original version, so no need to think about this.

File: ipt_connect/IPTdev/models.py (hunk 2 above)
Comment (nickkolok) [reply ipt_connect/IPTdev/models.py]: Fixed: https://github.com/nickkolok/ipt_connect/commit/ebe1b1abbeb2d89f5b5bc3c9ddb6c1f7ee967a5f

**Pull request merged**